### PR TITLE
Support carriage returns, tabs, and "[0;m"

### PR DESCRIPTION
**Pull request by zaquestion**
This is in an effort to support cleaning the GitLab CI trace files returned from their API. The regex change is done to support "[0;m" tough to find docs but seemingly identical to "[0m"

Note: This duplicates some changes in #6 -- in testing just the carriage return handling was suitable for gitlab ci trace file cases

### CODE REVIEW

File: vtclean.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 )
 
 // see regex.txt for a slightly separated version of this regex
-var vt100re = regexp.MustCompile(`^\033([\[\]]([\d\?]+)?(;[\d\?]+)*)?(.)`)
+var vt100re = regexp.MustCompile(`^\033([\[\]]([\d\?]+)?(;[\d\?]*)*)?([m@GCDPK;l])`)
```
Comment (zaquestion) [inline vtclean.go:10]: I'm honestly not thrilled about `([m@GCDPK;l])`, however leaving it as `(.)` broke this test as the `r` would get matched
```
"ccc \033]4;1;rgb:38/54/71test":       "ccc rgb:38/54/71test",
```

**Comment by lunixbochs** [comment]
Can you send me an example CI trace that needed these changes to work so I can think about cleaning this up?

**Comment by zaquestion** [comment]
Oops, sorry put this up late meant to include one [out.txt](https://github.com/lunixbochs/vtclean/files/2124839/out.txt)

File: vtclean_test.go [hunk 2]
```diff
@@ -43,10 +43,22 @@ var tests = map[string]string{
...
+	// carriage return
+	"aaa\rb": "baa",
 }
 
 var colorTests = map[string]string{
 	"aaa \033[25;25mtest": "aaa \033[25;25mtest\x1b[0m",
+
+	// [0;m cases
+	// TODO: trailing [0;m should probably not be overridden with [0m
+	// TODO: when [0;m seen, may not want to append [0m to the end
```
Comment (lunixbochs) [inline vtclean_test.go:59]: can you explain these TODOs?

**Comment by lunixbochs** [comment]
Merged manually

**Comment by lunixbochs** [comment]
I ended up diving into ECMA-48 and coming up with a much stronger primary regex. Thanks for the PR!

I also might use `lab`, it seems handy :)

**Pull request merged**